src/home/homework-luks.c

```diff
@@ -2316,9 +2317,17 @@ int home_create_luks(
                 log_info("Allocating image file completed.");
         }
 
+        if ( h->luks_sector_size == UINT64_MAX) {
+                /* if sector size is not specified, select UINT32_MAX, i.e. auto-probe */
+                image_sector_size = UINT32_MAX;
```

Comment:
man, this type confusion everywhere is confusing. libcryptsetup wants 32bit, libfdisk wants somathing opaque. I am pretty sure we should just use uint64_t everywhere and convert to the right types at the latest moment possible.

hence I think we should really use UINT64_MAX here for probing, and then inside of image_sector_size() map that to UINT32_MAX (and refuse any values > UINT32_MAX and < UINT64_MAX

Reply:
ah, christ, our own loop_device_make() already gets that wrong... grmbl. i guess it's fine then to leave that cleanup to another day